src/bp/core/services/ghost/cache-invalidators.ts

```diff
@@ -72,9 +72,10 @@ export namespace CacheInvalidators {
         return
       }
 
-      const relativePath = forceForwardSlashes(path.relative(process.PROJECT_LOCATION, path.dirname(file)))
-      this.cache.events.emit('invalidation', relativePath)
-      await this.cache.invalidateStartingWith(relativePath)
+      const relativePath = forceForwardSlashes(path.relative(process.PROJECT_LOCATION, file))
+      // TODO : why invalidate the whole folder?
+      await this.cache.invalidateStartingWith(path.dirname(relativePath))
+      this.cache.events.emit('userFileUpdate', relativePath)
```

Comment:
So I added this new event here to let me handle file changes indepentently from what happens in the `ObjectCache`. This `userFileUpdate` event should probably made accessible from somewhere else than the cache, since it's not a cache event at all.